internal/output/result.go

```diff
@@ -15,6 +17,27 @@ type pkgWithSource struct {
 // Custom implementation of this unique set map to allow it to serialize to JSON
 type pkgSourceSet map[pkgWithSource]struct{}
 
+// StableKeys returns the pkgWithSource keys in a deterministic order
+func (pss *pkgSourceSet) StableKeys() []pkgWithSource {
+	pkgWithSrcKeys := maps.Keys(*pss)
+
+	slices.SortFunc(pkgWithSrcKeys, func(a, b pkgWithSource) int {
+		for _, fn := range []func() int{
+			func() int { return strings.Compare(a.Source.Path, b.Source.Path) },
+			func() int { return strings.Compare(a.Package.Name, b.Package.Name) },
+			func() int { return strings.Compare(a.Package.Version, b.Package.Version) },
+		} {
```

Comment:
I'm super on the fence about this: on the one hand its clever, but on the other maybe it's too clever?

I've got a commit locally that unwinds this into `if` statements which I can push up if folks agree it's too clever

Reply:
I think it's easier to parse than if statements once you understand what it's doing, can you just add a comment above it basically describing it's just sorting by each field in descending priority.

Reply:
sweet done - one other annoying downside though is that we can't cover the `return 0` 😞

Reply:
Ok turns out there's a `cmp.Or` for exactly this - it's technically less efficient because it's not lazy ([though there's a proposal for that](https://github.com/golang/go/issues/66986)), but these compares should be pretty inexpensive and its nice to remove the unreachable code path